services/blockstorage/test/get_block_headers.go

```diff
@@ -0,0 +1,105 @@
+package test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/orbs-network/orbs-network-go/test"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+)
+
+var _ = Describe("Block storage", func () {
+	When("asked to provide transactions block header", func () {
+		It("returns transactions block header", func () {
+			driver := NewDriver()
+			driver.expectCommitStateDiff()
+
+			block := test.BlockPairBuilder().Build()
+			driver.commitBlock(block)
+
+			output, err := driver.blockStorage.GetTransactionsBlockHeader(&services.GetTransactionsBlockHeaderInput{BlockHeight:1})
+
+			Expect(err).ToNot(HaveOccurred())
+			Expect(output.TransactionsBlockHeader).To(Equal(block.TransactionsBlock.Header))
```

Comment:
make sure that your persistence adapter (in-memory) does not hold a pointer to the block but rather clones its bytes, so this equality check actually checks content and not just pointers